spec/helpers/date_helper_spec.rb

```diff
@@ -41,32 +41,56 @@
 
   describe "#parse_usa_date" do
 
-    context "passed bad dates" do
-      ["05012012", "somegarbage", "11/31/2012", nil, "9/"].each do |bad_date_string|
-        it "should not raise error for: #{bad_date_string}" do
+    context "given bad dates" do
+      ["05012012", "somegarbage", "11/31/2012", nil, "9/", "5/1/20114", "1/4/16"].each do |bad_date_string|
+        it "does not raise error for: #{bad_date_string}" do
           expect { parse_usa_date(bad_date_string) }.not_to raise_error
         end
 
-        it "should return nil for #{bad_date_string}" do
+        it "returns nil for #{bad_date_string}" do
           expect(parse_usa_date(bad_date_string)).to be_nil
         end
       end
     end
 
-    context "passed valid dates" do
-      it "parse a usa formatted date properly" do
+    context "given valid dates" do
+      it "parses a date with month and day in USA style" do
         expect(parse_usa_date("05/10/2012")).to eq(Time.zone.parse("2012-05-10"))
       end
 
       it "parses a usa formatted date with single digits properly" do
         expect(parse_usa_date("5/1/2012")).to eq(Time.zone.parse("2012-05-01"))
       end
 
-      it "should truncate a date with more than four digits in year" do
-        expect(parse_usa_date("5/1/20114")).to eq(Time.zone.parse("2011-05-01"))
+      it "can handle leading/trailing whitespace" do
+        expect(parse_usa_date("  5/1/2012  ")).to eq(Time.zone.parse("2012-05-01"))
       end
+    end
+
+    it "returns the datetime when given a Date" do
+      date = Date.new(2012, 5, 1)
+      expect(parse_usa_date(date)).to eq(Time.zone.parse("2012-05-01"))
+    end
 
+    it "returns the datetime when given a Date and time string" do
+      date = Date.new(2012, 5, 1)
+      expect(parse_usa_date(date, "3:17 PM")).to eq(Time.zone.parse("2012-05-01 15:17"))
     end
+
+    context "with a time as well" do
+      it "parses a date with month and day in USA style" do
+        expect(parse_usa_date("05/10/2012", "3:17 PM")).to eq(Time.zone.parse("2012-05-10 15:17"))
+      end
+
+      it "returns nil  with invalid time" do
+        expect(parse_usa_date("5/10/2012", "34:23 FM")).to be_nil
+      end
+
+      it "returns nil for valid time, but nil input" do
+        expect(parse_usa_date(nil, "3:17 PM")).to be_nil
+      end
+    end
```

Comment:
Should there be a specific test for the `date =~ /\A\d{4}-\d{2}-\d{2}/` condition or am I missing it?